lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -571,22 +696,38 @@ final HybridTextEditing textEditing = HybridTextEditing();
 /// - HTML's contentEditable feature handles typing and text changes.
 /// - HTML's selection API handles selection changes and cursor movements.
 class HybridTextEditing {
-  /// The default HTML element used to manage editing state when a custom
-  /// element is not provided via [useCustomEditableElement].
-  TextEditingElement _defaultEditingElement;
+  /// The text editing stategy used. It can change depending on the
+  /// formfactor/browser.
+  ///
+  /// It uses an HTML element to manage editing state when a custom element is
+  /// not provided via [useCustomEditableElement]
+  DefaultTextEditingStrategy _defaultEditingElement;
 
   /// Private constructor so this class can be a singleton.
+  ///
+  /// The constructor also decides which text editing strategy to use depending
+  /// on the operating system and browser engine.
   HybridTextEditing() {
-    _defaultEditingElement = TextEditingElement(this);
+    if (browserEngine == BrowserEngine.webkit &&
+        operatingSystem == OperatingSystem.iOs) {
+      this._defaultEditingElement = IOSTextEditingStrategy(this);
+    } else if (browserEngine == BrowserEngine.blink &&
+        operatingSystem == OperatingSystem.android) {
+      this._defaultEditingElement = AndroidTextEditingStrategy(this);
+    } else if (browserEngine == BrowserEngine.firefox) {
+      this._defaultEditingElement = FirefoxTextEditingStrategy(this);
+    } else {
+      this._defaultEditingElement = DefaultTextEditingStrategy(this);
+    }
   }
 
   /// The HTML element used to manage editing state.
   ///
   /// This field is populated using [useCustomEditableElement]. If `null` the
-  /// [_defaultEditableElement] is used instead.
-  TextEditingElement _customEditingElement;
+  /// [_defaultEditingElement] is used instead.
+  DefaultTextEditingStrategy _customEditingElement;
```

Comment:
several variables/methods should be renamed from editing element to editing strategy.